README.md

```diff
@@ -1,81 +1,145 @@
+sudo pkg install zstd
 ```
 
-## Quick way
+### Install QEMU emulator
 
-### You can use pre-built images, otherwise proceed to next step.
 ```
-fetch https://artifact.ci.freebsd.org/snapshot/head/latest/riscv/riscv64/bbl.xz
-unxz bbl.xz
+sudo pkg install qemu-devel
+```
+
+### Download FreeBSD/RISC-V snapshot and kernel
+
+```
+fetch https://artifact.ci.freebsd.org/snapshot/head/latest/riscv/riscv64/disk-test.img.zst
+zstd -d disk-test.img.zst -o riscv.img
+fetch https://artifact.ci.freebsd.org/snapshot/head/latest/riscv/riscv64/kernel.txz 
+tar Jxvf kernel.txz --strip-components 3 boot/kernel/kernel
+```
+
+### Configure tap(4) 
+
+Load (tun)tap module(s) if not already loaded
+
+If FreeBSD >= 13:
+
+```
+sudo kldload if_tuntap
+```
+
+Else:
+
+```
+sudo kldload if_tun if_tap
+```
+
+Configure the network with tap(4):
+
+```
+ifconfig tap0 create up
+ifconfig bridge0 create up
+ifconfig bridge0 addm em0 addm tap0 # em0 might depends of your ethernet devices
+```
+
+### Run FreeBSD/RISC-V in QEMU
+
+```
+sudo qemu-system-riscv64 -machine virt -m 2048M -smp 2 -nographic -kernel $HOME/riscv/kernel -bios /usr/local/share/opensbi/lp64/generic/firmware/fw_jump.elf -drive file=$HOME/riscv/riscv.img,format=raw,id=hd0 -device virtio-blk-device,drive=hd0 -netdev tap,ifname=tap0,script=no,id=net0 -device virtio-net-device,netdev=net0
+# login as root without password
 ```
 
 ## Complete build from scratch
 Set the following environment variables:
+
+with (t)csh shell:
+
 ```
-$ setenv MAKEOBJDIRPREFIX /home/${USER}/obj/
-$ setenv WITHOUT_FORMAT_EXTENSIONS yes
-$ setenv DESTDIR /home/${USER}/riscv-world
+setenv MAKEOBJDIRPREFIX ${HOME}/riscv/obj/
+setenv DESTDIR ${HOME}/riscv/riscv-world
 ```
 
-### Build FreeBSD world
+with bash/zsh/etc. shell:
+
 ```
-$ svnlite co http://svn.freebsd.org/base/head freebsd-riscv
-$ cd freebsd-riscv
-$ make -j4 CROSS_TOOLCHAIN=riscv64-gcc TARGET_ARCH=riscv64 buildworld
+export MAKEOBJDIRPREFIX=${HOME}/riscv/obj/
+export DESTDIR=${HOME}/riscv/riscv-world
 ```
 
-### Build 32mb rootfs image
+### Build FreeBSD/RISC-V
+
+Get source from svn:
 ```
-$ cd freebsd-riscv
-$ make TARGET_ARCH=riscv64 -DNO_ROOT -DWITHOUT_TESTS DESTDIR=$DESTDIR installworld
-$ make TARGET_ARCH=riscv64 -DNO_ROOT -DWITHOUT_TESTS DESTDIR=$DESTDIR distribution
-$ fetch https://raw.githubusercontent.com/bukinr/riscv-tools/master/image/basic.files
-$ tools/tools/makeroot/makeroot.sh -s 32m -f basic.files riscv.img $DESTDIR
+svnlite co http://svn.freebsd.org/base/head ${HOME}/riscv/freebsd-riscv
 ```
 
-### Prepare your kernel config
-Modify sys/riscv/conf/GENERIC. Uncomment the following lines and specify the path to your riscv.img:
+Build FreeBSD/RISC-V
 ```
-options 	MD_ROOT
-options 	MD_ROOT_SIZE=32768	# 32MB ram disk
-makeoptions	MFS_IMAGE=/path/to/riscv.img
-options 	ROOTDEVNAME=\"ufs:/dev/md0\"
+cd ${HOME}/riscv/freebsd-riscv
+make -j4 TARGET_ARCH=riscv64 buildworld
+make -j4 TARGET_ARCH=riscv64 buildkernel
```

Comment:
Maybe mention that the parameter given to -j can be output of `sysctl -n kern.smp.cpus`?